industry_benchmarks/utils/results_cleanup.py

```diff
@@ -0,0 +1,349 @@
+    """
+    analyzer.use_full_trajectory = True
+    u_ln, N_l = analyzer._compute_mbar_decorrelated_energies()
+    return u_ln, N_l
+
+
+def get_replica_state_indices(analyzer):
+    energy_data = list(analyzer._read_energies(truncate_max_n_iterations=True))
+    replicas_state_indices = energy_data[-1]
+    return replicas_state_indices
+
+
+def subsample_traj(simulation, hybrid_system_pdb, lambda_windows, outfile):
+    for i in range(0, lambda_windows):
+        u = mda.Universe(hybrid_system_pdb, simulation, format=FEReader, state_id=i)
+        frames = [round(i) for i in np.linspace(0, len(u.trajectory) - 1, 21)]
+        out_traj = pathlib.Path(f"{outfile}_{i}.xtc")
+        with mda.Writer(str(out_traj), n_atoms=len(u.atoms)) as w:
+            for ts in u.trajectory[frames]:
+                w.write(u.atoms)
+
+
+def extract_data(simulation, checkpoint, hybrid_pdb, outfile, out_traj="out"):
+    """
+    Extract the MBAR-ready energy matrix and replica state indices.
+
+    Also extracts some info about the simulation and saves info in a
+    info.yaml file in the same folder as out_traj
+
+    Parameters
+    ----------
+    simulation: pathlib.Path
+        Path to the simulation `.nc` file
+    checkpoint: pathlib.Path
+        Path to the checkpoint `.chk` file
+    hybrid_pdb: pathlib.Path
+        Path to the `.pdb` file of the hybrid system
+    outfile: pathlib.Path
+        Path to the output `.npz` file
+    out_traj: pathlib.Path
+        Path to the output `.xtc` files. A separate file is created for every
+        lambda window. The state number is appended to the filename. Default: 'out'
+    """
+    if not simulation.is_file() or not checkpoint.is_file():
+        errmsg = "Either the simulation or checkpoint file could not be found"
+        raise ValueError(errmsg)
+
+    reporter = multistate.MultiStateReporter(
+        storage=simulation.as_posix(),
+        open_mode="r",
+        checkpoint_storage=checkpoint.as_posix(),
+    )
+
+    analyzer = multistate.MultiStateSamplerAnalyzer(reporter)
+    u_ln, N_l = compute_mbar_energies(analyzer)
+    replicas_state_indices = get_replica_state_indices(analyzer)
+    np.savez(outfile, u_ln=u_ln, N_l=N_l, replicas_state_indices=replicas_state_indices)
+
+    lambda_windows = len(replicas_state_indices)
+    subsample_traj(simulation, hybrid_pdb, lambda_windows, out_traj)
+
+    # get some info and save it in info.yaml
+    print("Creating info.yaml")
+    n_atoms = reporter._storage_checkpoint.dimensions["atom"].size
+
+    with open(
+        simulation.parent / "simulation_real_time_analysis.yaml", "r"
+    ) as yaml_file:
+        real_time_analysis_data = yaml.safe_load(yaml_file)
+        # grab the last entry
+        ns_per_day = real_time_analysis_data[-1]["timing_data"]["ns_per_day"]
+
+    with open(out_traj.parent / "info.yaml", "w") as yaml_file:
+        yaml.dump({"n_atoms": n_atoms, "ns_per_day": ns_per_day}, yaml_file)
+
```

Comment:
If you call this a bunch of time I think we'll get issues with reporters being kept open.

How about we add:

```suggestion

    analyzer.close()
    reporter.close()

```

Reply:
Ah yes if they need that then we leave the files open until the script is done